src/renderer/store/modules/playlists.js

```diff
@@ -1,32 +1,83 @@
+import crypto from 'crypto'
 import { DBPlaylistHandlers } from '../../../datastores/handlers/index'
 
+function generateRandomPlaylistId() {
+  return `ft-playlist--${generateRandomUniqueId()}`
+}
+
+function generateRandomPlaylistName() {
+  return `Playlist ${new Date().toISOString()}-${Math.floor(Math.random() * 10000)}`
+}
+
+function generateRandomUniqueId() {
+  return crypto.randomUUID()
+}
+
 const state = {
-  playlists: [
+  // Playlist loading takes time on app load (new windows)
+  // This is necessary to let components to know when to start data loading
+  // which depends on playlist data being ready
+  playlistsReady: false,
+  playlists: [],
+  defaultPlaylists: [
     {
       playlistName: 'Favorites',
-      protected: true,
-      videos: []
+      protected: false,
+      description: 'Your favorite videos',
+      videos: [],
+      _id: 'favorites',
     },
     {
-      playlistName: 'WatchLater',
-      protected: true,
-      removeOnWatched: true,
-      videos: []
-    }
-  ]
+      playlistName: 'Watch Later',
```

Comment:
Discussion on this: It seems like the current Favorites functionality in FreeTube today is the equivalent of a Watch Later. Why is this being separated in this manner? More broadly, I want to pick your brain on what your plans for this are. My top concerns for this PR are:
1. Maintaining reverse compatibility for users who download the new version but want to revert for whatever reason
2. Adding back all of the removed instances of the favorite icon in the JS, template, and CSS files with a functionally equivalent or better Add to Watch Later button. I think having a release without this present would be a strong negative change, as our dedicated users are the ones who have adapted to saving videos in this one-click way we have set up. Because of the sheer importance of this, I want to make sure that your plans for this are made clear for both the sake of discussion and for allowing other devs to follow through with this in a way that's aligned with your vision if you were unable to complete development on this feature.

Reply:
@PikachuEXE thoughts on these two points ?(in case you missed the notification)

Reply:
> It seems like the current Favorites functionality in FreeTube today is the equivalent of a Watch Later.

It might be used that way (or not, since I don't use it as watch later ever), since there is **only one playlist** before this PR.

1. Reverting is gonna be moving (copying is fine) all videos into **the first playlist** (well current version of FT just display the list of videos of first playlist, you can check exported playlist files). It's not automatic but if you want a version which has automatic revert function we will need to release a version just for that (before releasing a version with this PR)
2. I would NOT add more functionality to this PR (which is already a complicated one). The current UX is based on having **one playlist** only and I would not consider that is good for UX with privacy (over one-click features). Feel free to develop and propose another PR if anyone is willing to.